File: elixir/bob/bob.exs
defmodule Teenager do
  def hey (input) do
    cond do
      String.ends_with?(input, "?") -> "Sure."
      String.match?(input, ~r/^\s*$/) -> "Fine. Be that way!"
      String.match?(input, ~r/^[^[:lower:]]*[[:upper:]][^[:lower:]]*$/) ->
        "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: python/bob/bob.py
def hey(what):
  what = what.strip()
  if what.upper() == what and what.lower() != what:
    return "Whoa, chill out!"
  elif what == "":
    return "Fine. Be that way!"
  elif what[-1] == '?':
    return "Sure."
  else:
    return "Whatever."
